src/transport_ofi.c

```diff
@@ -603,24 +610,44 @@ static inline int exchange_and_av_insert(struct fabric_info *info)
 	return ret;
     }
 
+    full_ep_len = full_ep_len + epnamelen;
+    epnamebuff = malloc(full_ep_len);
+
     alladdrs = malloc(info->npes * epnamelen);
     if(alladdrs==NULL){
 	OFI_ERRMSG("alladdrs is NULL\n");
 	return ret;
     }
 
-    ret = shmem_runtime_put("OFI", epname, epnamelen);
+#ifdef USE_ON_NODE_COMMS
+    gethostname(&epname[epnamelen], pidlen);
+#endif
+
+    ret = shmem_runtime_put("OFI", epname, full_ep_len);
```

Comment:
Can we use a more specific string here, to avoid a potential future clash?  For example, "fi_hostname", following the convention of other OFI entries in the PMI KVS.